packages/lesswrong/components/common/TabNavigationMenu/SubforumsList.tsx

```diff
@@ -19,6 +19,16 @@ const styles = ((theme: ThemeType): JssStyles => ({
     height: 'unset',
     minHeight: 24,
   },
+  maybeHiddenResultsHidden: {
+    overflow: 'hidden',
+    maxHeight: 0,
+    transition: 'max-height 0.3s ease-in-out',
+  },
+  maybeHiddenResultsVisible: {
+    overflow: 'hidden',
+    maxHeight: 1000,
+    transition: 'max-height 0.3s ease-in-out',
```

Comment:
I could only get this transition to work on expand, not on collapse. Any tips would be appreciated

Reply:
Probably something to do with React getting confused about whether or not it's the same DOM element between renders. In general, these kind of transitions are super easy to break in React when doing them manually so you'd probably be better off doing something like:
```
import Collapse from "@material-ui/core/Collapse";

...

          <Collapse in={showAll}>
            <div>
              {maybeHiddenResults.map((subforum) => getListItem(subforum))}
            </div>
          </Collapse>

```

It's not as efficient, but it's a lot less error prone (both now, and when making changes in the future).

Reply:
I think it's because the contents of the expanded part get removed immediately, so you don't see an animation. It works when I remove the `showAll` check from `{showAll && maybeHiddenResults.map((subforum) => getListItem(subforum))}`. Unfortunately it appears as if there's a delay in the closing animation because it's animating from a large `max-height`, so I think this isn't the best solution in a case like this where the height of the contents can vary a lot. I haven't used `Collapse` but prob worth trying that instead.